app/components/layertimeline/layertimeline.js

```diff
@@ -46,6 +46,14 @@ class LayerTimelineController {
       }
     }, $scope);
 
+    this.studioState_.onChange((event, changes) => {
```

Comment:
this entire block should no longer be needed if you `this.scope_.$apply` above. the basic idea is that the changes to studioState above are done inside Angular's digest cycle, so after the selection is set, anything that's bound to the current selection (like the layer timeline items) automatically update